File: typescript-angular/API/Client/CloudUserApi.ts
/// <reference path="api.d.ts" />

/* tslint:disable:no-unused-variable member-ordering */

namespace API.Client {
    'use strict';

    export class CloudUserApi {
        protected basePath = 'http://api.cloudboost.io/';
        public defaultHeaders : any = {};

        static $inject: string[] = ['$http', '$httpParamSerializer'];

        constructor(protected $http: ng.IHttpService, protected $httpParamSerializer?: (d: any) => any, basePath?: string) {
            if (basePath) {
                this.basePath = basePath;
            }
        }

        private extendObj<T1,T2>(objA: T1, objB: T2) {
            for(let key in objB){
                if(objB.hasOwnProperty(key)){
                    objA[key] = objB[key];
                }
            }
            return <T1&T2>objA;
        }

        /**
         * add a user to a role
         * 
         * @param appId ID of app
         * @param body request body
         */
        public addToRole (appId: string, body: UserRoleBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/addToRole'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling addToRole');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling addToRole');
            }
            let httpRequestParams: any = {
                method: 'PUT',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * change a password
         * changes the password for a logged in user without logging them out
         * @param appId ID of app
         * @param body request body
         */
        public changePassword (appId: string, body: PasswordChangeBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/changePassword'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling changePassword');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling changePassword');
            }
            let httpRequestParams: any = {
                method: 'PUT',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * login existing user
         * 
         * @param appId ID of app
         * @param body request body
         */
        public login (appId: string, body: UserBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/login'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling login');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling login');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * logout user
         * 
         * @param appId ID of app
         * @param body request body
         */
        public logout (appId: string, body: UserBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/logout'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling logout');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling logout');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * remove a user from a role
         * 
         * @param appId ID of app
         * @param body request body
         */
        public removeFromRole (appId: string, body: UserRoleBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/removeFromRole'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling removeFromRole');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling removeFromRole');
            }
            let httpRequestParams: any = {
                method: 'PUT',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * reset a password
         * sends a password reset link to the given email
         * @param appId ID of app
         * @param body request body
         */
        public resetPassword (appId: string, body: EmailBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<string> {
            const localVarPath = this.basePath + 'user/{app_id}/resetPassword'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling resetPassword');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling resetPassword');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * create a new user
         * 
         * @param appId ID of app
         * @param body request body
         */
        public signUp (appId: string, body: UserBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudUser> {
            const localVarPath = this.basePath + 'user/{app_id}/signup'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling signUp');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling signUp');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
    }
}

File: akka-scala/src/main/scala/io/swagger/client/api/CloudUserApi.scala
package io.swagger.client.api

import io.swagger.client.model.UserRoleBody
import io.swagger.client.model.CloudUser
import io.swagger.client.model.PasswordChangeBody
import io.swagger.client.model.UserBody
import io.swagger.client.model.EmailBody
import io.swagger.client.core._
import io.swagger.client.core.CollectionFormats._
import io.swagger.client.core.ApiKeyLocations._

object CloudUserApi {

  /**
   * 
   * Expected answers:
   *   code 200 : CloudUser (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def addToRole(appId: String, body: UserRoleBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.PUT, "http://api.cloudboost.io/", "user/{app_id}/addToRole", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      
  /**
   * changes the password for a logged in user without logging them out
   * 
   * Expected answers:
   *   code 200 : CloudUser (password reset)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def changePassword(appId: String, body: PasswordChangeBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.PUT, "http://api.cloudboost.io/", "user/{app_id}/changePassword", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      
  /**
   * 
   * Expected answers:
   *   code 200 : CloudUser (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def login(appId: String, body: UserBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.POST, "http://api.cloudboost.io/", "user/{app_id}/login", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      
  /**
   * 
   * Expected answers:
   *   code 200 : CloudUser (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def logout(appId: String, body: UserBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.POST, "http://api.cloudboost.io/", "user/{app_id}/logout", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      
  /**
   * 
   * Expected answers:
   *   code 200 : CloudUser (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def removeFromRole(appId: String, body: UserRoleBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.PUT, "http://api.cloudboost.io/", "user/{app_id}/removeFromRole", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      
  /**
   * sends a password reset link to the given email
   * 
   * Expected answers:
   *   code 200 : String (password reset)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def resetPassword(appId: String, body: EmailBody): ApiRequest[String] =
    ApiRequest[String](ApiMethods.POST, "http://api.cloudboost.io/", "user/{app_id}/resetPassword", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[String](200)
      
  /**
   * 
   * Expected answers:
   *   code 200 : CloudUser (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def signUp(appId: String, body: UserBody): ApiRequest[CloudUser] =
    ApiRequest[CloudUser](ApiMethods.POST, "http://api.cloudboost.io/", "user/{app_id}/signup", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudUser](200)
      


}

